numba/targets/codegen.py

```diff
@@ -387,9 +389,63 @@ def get_pointer_to_function(self, name):
         return self._codegen._engine.get_function_address(name)
 
     def _finalize_specific(self):
+        self._codegen._scan_and_fix_unresolved_refs(self._final_module)
         self._codegen._engine.finalize_object()
 
 
+class RuntimeLinker(object):
+    """
+    For tracking unresolved symbols generated at runtime due to recursion.
+    """
+    PREFIX = '.numba.unresolved$'
```

Comment:
it's used in https://github.com/numba/numba/pull/2130/files#diff-ec37b531d0f5e6e4ff2bb91a63bd651dR588 to define unresolved symbols.